bin/kintsugi

```diff
@@ -10,43 +10,101 @@ require "optparse"
 require "kintsugi"
 require_relative "../lib/kintsugi/version"
 
+Command = Struct.new(:option_parser, :action, :description, keyword_init: true)
 
-def parse_options!(argv)
-  options_parser = create_options_parser
-
-  options = {}
-  options_parser.parse!(argv, into: options)
-
-  if argv.length != 1
-    puts "Incorrect number of arguments\n\n"
-    puts options_parser
-    exit(1)
-  end
-
-  options
-end
-
-def create_options_parser
+DRIVER_OPTION_PARSER =
   OptionParser.new do |opts|
-    opts.banner = "Kintsugi, version #{Kintsugi::Version::STRING}\nCopyright (c) 2021 " \
-                  "Lightricks\n\nUsage: kintsugi [pbxproj_filepath] [options]"
-    opts.on("--changes-output-path=PATH", "Path to which changes applied to the project are " \
-            "written in JSON format. Used for debug purposes.")
+    opts.banner = "Usage: kintsugi driver BASE OURS THEIRS ORIGINAL_FILE_PATH\n" \
+      "Uses Kintsugi as a Git merge driver. Parameters " \
+      "should be the path to base version of the file, path to ours version, path to theirs " \
+      "version, and the original file path."
 
     opts.on("-h", "--help", "Prints this help") do
       puts opts
       exit
     end
+  end
 
-    opts.on_tail("-v", "--version", "Prints version") do
-      puts Kintsugi::Version::STRING
-      exit
+DRIVER_ACTION =
+  lambda { |_, arguments, option_parser|
+    if arguments.count != 4
+      puts "Incorrect number of arguments to 'driver' subcommand\n\n"
+      puts option_parser
+      exit(1)
     end
+    Kintsugi.three_way_merge(arguments[0], arguments[1], arguments[2], arguments[3])
+  }
+
+COMMANDS = {
+  "driver" => Command.new(
+    option_parser: DRIVER_OPTION_PARSER,
+    action: DRIVER_ACTION,
+    description: "Git merge driver mode"
```

Comment:
IMO, a subcommand is not a mode, I would go with a verb `3-way merge compatible with Git merge driver`
or similar.